src/go/readme.md

```diff
@@ -0,0 +1,217 @@
+
+Finally, the sample also demonstrates how to use the different [types of managed identities](https://learn.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview#managed-identity-types) to get an access token.
+
+For more information about how the protocols work in this scenario and other scenarios, see [Authentication Scenarios for Azure AD](http://go.microsoft.com/fwlink/?LinkId=394414).
+
+For more information about Managed Identity, please visit the [Managed Identities for Azure Resources homepage](https://learn.microsoft.com/azure/active-directory/managed-identities-azure-resources/overview).
+
+## Topology
+
+<img alt="Overview" src="./images/call-kv.png" />
+
+### Scenario
+
+You want to access an Azure Key Vault secret from a function deployed in Azure. And, you don't want to worry about managing secrets or app credentials.
+
+## How To Run This Sample
+
+To run this sample, you'll need:
+
+- An IDE of your choice, for example [Visual Studio Code](https://code.visualstudio.com/download)
+- An Internet connection
+- An Azure account to create, deploy, and manage applications. If you do not have an Azure Account, follow the [instructions](https://azure.microsoft.com/free/) to get a free account.
+
+### Step 1:  Clone or download this repository
+
+From your shell or command line:
+
+```Shell
+git clone https://github.com/Azure-Samples/msal-managed-identity.git
+```
+
+or download and extract the repository `.ZIP` file.
+
+The GO sample is located in the [`/src/go/devapps/managedidentity`](https://github.com/AzureAD/microsoft-authentication-library-for-go/blob/c5febcbae287a26a0cfedd45f4edeaf3c41ad7dc/apps/tests/devapps/managedidentity/managedidentity_sample.go) folder.
+
+### Step 2:  Modify the Key Vault URI and Secret name values in the code
+
+Following are the changes you need to make:
+
+- In the [`handler.go`](https://github.com/Azure-Samples/msal-managed-identity/blob/main/src/go/sample/AcquireTokenMSI.go) file under the getSecretFromAzureVault method modify the following values, 
+
+    ```go
+        keyVaultUri := "your-key-vault-uri"
+        secretName := "your-secret-name"
+    ```
+
+- Change these to match your key vault uri and secret name. These can be found in the following locations:
+
+1. Key Vault URI - In your Azure home page, go to your key vault, on the Overview page our key vault URI can be found under **'Essentials'**
+1. Secret Name - On the Key Vault Overview page, go to the Panel on the left and expand the **'Objects'** dropdown  
+Click into **'Secrets'**  
+Click into the secret you want to use  
+Click ont the version you would like to use  
+Copy the part after the key vault URI and use that as your secret name  
+
+## Publish your function
```

Comment:
Since the app service is not implemented, will it be relevant to have a sample for the functions? Can we have a sample for VM instead since that is implemented?

Reply:
I marked this as a draft PR so its ready to go when App Service is implemented. The other docs in this sample project are using AppService so I think we want to keep this the same, and then the other PR that is open is more about using the library